aws/resource_aws_codebuild_project.go

```diff
@@ -59,6 +59,24 @@ func resourceAwsCodeBuildProject() *schema.Resource {
 				},
 				Set: resourceAwsCodeBuildProjectArtifactsHash,
 			},
+			"cache": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validateAwsCodeBuildCacheType,
```

Comment:
Nitpick: This can be simplified with:
```go
ValidateFunc: validation.StringInSlice([]string{
  codebuild.CacheTypeNoCache,
  codebuild.CacheTypeS3,
}, false),
```